happi/cli.py

```diff
@@ -396,21 +396,48 @@ def edit(ctx, name: str, edits: list[str]):
 
 
 @happi_cli.command()
-@click.argument('item_names', nargs=-1)
+@click.option('--glob/--regex', 'use_glob', default=True,
+              help='use glob style (default) or regex style search terms. '
+              r'Regex requires backslashes to be escaped (eg. at\\d.\\d)')
+@click.argument('search_criteria', nargs=-1)
 @click.pass_context
-def load(ctx, item_names: list[str]):
-    """Open IPython terminal with ITEM_NAMES loaded."""
+def load(
+    ctx: click.Context,
+    use_glob: bool,
+    search_criteria: list[str]
+):
+    term = []
+    results = set()
+
+    for item in search_criteria:
+        term.append(item)
```

Comment:
I peeked at this PR again because I saw the merge commits

I'm not sure I understand why we're accumulating the user's input like this, I think we might just want to pass `item` as the `search_criteria`.

I'm reading this loop as:
- first loop: search using only the first term
- second loop: search using the first and second terms together
- third loop: search using the first, second, and third terms together

But I think the desired behavior was:
- first loop: search using only the first term
- second loop: search using only the second term
- third loop: search using only the third term

Reply:
Scratch that, I didn't see the `term = []` at the end of the loop